frontend/packages/console-app/src/components/nodes/NodeLogs.tsx

```diff
@@ -0,0 +1,298 @@
+import './node-logs.scss';
+
+type NodeLogsProps = {
+  obj: NodeKind;
+};
+
+const NodeLogs: React.FC<NodeLogsProps> = ({ obj: node }) => {
+  const {
+    kind,
+    metadata: { labels, name, namespace: ns },
+    status,
+  } = node;
+  const isWindows = status?.nodeInfo.operatingSystem === 'windows';
+  const pathItems = ['journal'];
+  isWindows
+    ? pathItems.push('containers', 'hybrid-overlay', 'kube-proxy', 'kubelet')
+    : labels['node-role.kubernetes.io/master'] === '' &&
+      pathItems.push('openshift-apiserver', 'kube-apiserver', 'oauth-apiserver');
+  const pathQueryArgument = 'path';
+  const unitQueryArgument = 'unit';
+  const logQueryArgument = 'log';
+
+  const [path, setPath] = React.useState(getQueryArgument(pathQueryArgument) || pathItems[0]);
+  const [logURL, setLogURL] = React.useState('');
+  const [logFilenames, setLogFilenames] = React.useState([]);
+  const [unit, setUnit] = React.useState(getQueryArgument(unitQueryArgument));
+  const [logFilename, setLogFilename] = React.useState(getQueryArgument(logQueryArgument));
+  const [isLoadingLog, setLoadingLog] = React.useState(true);
+  const [isLoadingFilenames, setLoadingFilenames] = React.useState(true);
+  const [error, setError] = React.useState('');
+  const [isPathOpen, setPathOpen] = React.useState(false);
+  const [isFilenameOpen, setFilenameOpen] = React.useState(false);
+  const [content, setContent] = React.useState('');
+  const [isWrapLines, setWrapLines] = useUserSettings<boolean>(
+    `${USERSETTINGS_PREFIX}.log.wrapLines`,
+    false,
+    true,
+  );
+  const { t } = useTranslation();
+
+  const isJournal = path === 'journal';
+
+  const fetchLog = React.useCallback(
+    (url: string) => {
+      coFetch(url)
+        .then((response) => response.text())
+        .then((responseText) => {
+          setContent(responseText);
+          setLoadingLog(false);
+          setError('');
+        })
+        .catch((e) => {
+          setLoadingLog(false);
+          setError(t('public~Error fetching logs: {{message}}', { message: e.message }));
+        });
+    },
+    [t],
+  );
+  const getUnitQueryParams = (unitText: string) => {
+    const unitsArray = unitText?.split(',');
+    const unitQueryParams = unitsArray?.map((val) => `unit=${val}`);
+    return unitQueryParams.join('&');
+  };
+  const getLogURL = React.useCallback(
+    (ext?: string, unitText?: string) => {
+      const baseURL = `proxy/logs/${path}`;
+      let extendedURL;
+      if (ext) {
+        extendedURL = `${baseURL}${ext}`;
+      }
+      if (unitText) {
+        extendedURL = `${baseURL}?${getUnitQueryParams(unitText)}`;
+      }
+      return resourceURL(modelFor(kind), {
+        name,
+        ns,
+        path: extendedURL || baseURL,
+      });
+    },
+    [kind, name, ns, path],
+  );
+
+  React.useEffect(() => {
+    if (!path || isJournal) {
+      const journalLogURL = getLogURL('', unit);
+      setLogURL(journalLogURL);
+      fetchLog(journalLogURL);
```

Comment:
You could have a separate `useEffect` hook that calls `fetchLog` whenever `logURL` changes. Then you wouldn't need to call it each time.

```ts
React.useEffect(() => {
  if (logURL) {
    fetchLog(logURL);
  }
}, [logURL, fetchLog];
```